src/thunderbots/software/ai/hl/stp/play/play.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "ai/hl/stp/tactic/tactic.h"
+#include "ai/world/world.h"
+
+// Forward declaration of the Play class so the PlayFactory can reference it.
+class Play;
+
+/** Note **/
+// We keep the PlayFactory and Play classes in the same file to avoid cyclic includes
+// (where each file includes the other), since it causes compilation to fail
```

Comment:
This should be prevented via header guards no?

Reply:
Not in this case, because the order matters. I think as one file/class is being created it includes the other, which then includes the first one again. This is all before the corresponding classes are fully created, so the compiler doesn't know how to resolve the references.

Regardless I think the classes are small enough and related enough keeping them together is fine anyway, just wanted to be extra explicit about what I ran into as a warning for the future.